data-prepper-core/src/main/java/org/opensearch/dataprepper/acknowledgements/ExpiryMonitor.java

```diff
@@ -0,0 +1,48 @@
+package org.opensearch.dataprepper.acknowledgements;
+
+import org.opensearch.dataprepper.model.acknowledgements.ExpiryItem;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.concurrent.Executors;
+import java.util.concurrent.ScheduledExecutorService;
+import java.util.concurrent.ScheduledFuture;
+import java.util.concurrent.TimeUnit;
+
+public class ExpiryMonitor {
+    private static final Logger LOG = LoggerFactory.getLogger(ExpiryMonitor.class);
+
+    private static final long CANCEL_CHECK_INTERVAL_SECONDS = 15;
+    private static final ScheduledExecutorService SCHEDULED_EXECUTOR_SERVICE = Executors.newSingleThreadScheduledExecutor();
+    private static final ConcurrentHashMap<ExpiryItem, ScheduledFuture> EXPIRY_MONITORS = new ConcurrentHashMap<>();
+
+    public ExpiryMonitor() {
+        SCHEDULED_EXECUTOR_SERVICE.scheduleAtFixedRate(() -> {
```

Comment:
This class appears to be a singleton masquerading as a normal object.

I'd probably aim to make this an explicit singleton. 

Also, each call to `new` will run a new schedule. Is this correct?

I think the ideal experience for a caller is the following. It would be a clear singleton and avoid an unnecessary call to `new` and avoid adding a new object in the memory heap.

```
ExpiryMonitor.getInstance().addExpiryItem(expiryItem);
```

Reply:
That's a good call, it is intended to be a singleton